scripts/cleanup_unitstates.py

```diff
@@ -0,0 +1,226 @@
+    for r in raw:
+        model = r["_id"]["model"]
+        uuid = r["_id"]["uuid"]
+        secret = r["_id"]["secret"]
+        revs = r["revs"]
+        revs.sort()
+        active_secret_revisions[uuid][secret].extend(revs)
+    return active_secret_revisions
+
+
+def build_secret_states(opts):
+    """Read the mongo database, and return each unit's secret-obsolete-revisions.
+
+    :return: {"model-uuid": {"unit-id": secret-state}}
+    """
+    secret_states = collections.defaultdict(dict)
+    raw = read_secret_states(opts)
+    for state in raw:
+        id = state["_id"]
+        uuid = state["model-uuid"]
+        secret_state_str = state["secret-state"]
+        secret_state = yaml.safe_load(secret_state_str)
+        secret_states[uuid][id] = secret_state
+    return secret_states
+
+
+secret_prefix = "secret:"
+
+def print_missing_and_updated_unitstates(opts, active_secret_revisions, secret_states):
+    for uuid, states in secret_states.items():
+        model_secrets = active_secret_revisions[uuid]
+        for unit_id, state in states.items():
+            obsolete = state["secret-obsolete-revisions"]
+            new_obsolete = {}
+            has_changes = False
+            for secret_id, revisions in obsolete.items():
+                real_secret_id = secret_id
+                if secret_id.startswith(secret_prefix):
+                    real_secret_id = secret_id[len(secret_prefix):]
+                if real_secret_id not in model_secrets:
+                    print(f"// removing {unit_id} {real_secret_id}")
+                    continue
+                secret_revs = set(model_secrets[real_secret_id])
+                referenced_revs = set(revisions)
+                missing_revs = list(referenced_revs - secret_revs)
+                missing_revs.sort()
+                if not missing_revs:
+                    new_obsolete[secret_id] = revisions
+                    if opts.verbose:
+                        print(f"// skipping {unit_id} {real_secret_id}")
+                    continue
+                has_changes = True
+                remaining_revs = list(referenced_revs & secret_revs)
+                remaining_revs.sort()
+                print(f"// removing {unit_id} {real_secret_id} {missing_revs}")
+                new_obsolete[secret_id] = remaining_revs
+            if not has_changes:
+                if opts.verbose:
+                    print(f"// skipping {unit_id}")
+                continue
+            state["secret-obsolete-revisions"] = new_obsolete
+            as_str = yaml.dump(state).replace("\n", "\\n")
+            print(f'd.unitstates.update({{ "_id": "{unit_id}" }}, {{ "$set": {{ "secret-state": "{as_str}" }} }})')
+
+
+def main(args):
+    import argparse
+    p = argparse.ArgumentParser("simple script for removing a lot of secrets")
+    p.add_argument("--controller", default="0", type=str, help="Juju controller machine id")
```

Comment:
`--controller-id` ?
Since in the juju cli `--controller` takes the controller name and it might be confusing